xapian-glib/tests/database.c

```diff
@@ -134,6 +134,73 @@ database_writable_flags_no_termlist (void)
   delete_database ("glass-db");
 }
 
+static void
+database_writable_allterms (void)
+{
+  GError *error = NULL;
+  XapianWritableDatabase *wdb =
+    xapian_writable_database_new_full ("glass-db",
+                                       XAPIAN_DATABASE_ACTION_CREATE,
+                                       XAPIAN_DATABASE_BACKEND_GLASS,
+                                       0,
+                                       &error);
+
+  g_assert_nonnull (wdb);
+
+  g_object_add_weak_pointer (G_OBJECT (wdb), (gpointer *) &wdb);
+
+  XapianDocument *doc = xapian_document_new ();
+
+  g_object_add_weak_pointer (G_OBJECT (doc), (gpointer *) &doc);
+
+  xapian_document_add_term (doc, "one");
+  xapian_document_add_term (doc, "two");
+
+  xapian_writable_database_add_document (wdb, doc, NULL, &error);
+  g_object_unref (wdb);
+
+  g_object_unref (doc);
+
+  XapianDatabase *db =
+    xapian_database_new_with_path ("glass-db", &error);
+
+  g_object_add_weak_pointer (G_OBJECT (db), (gpointer *) &db);
```

Comment:
Aha, it seems the place I copied it from also tested `g_assert_null (db);` after `g_object_unref (db);`, so as you suggest it's verifying nothing leaks the object.

I'll add such assertions instead - it seems useful to have these extra checks, and it makes this new testcase more consistent with the existing ones.  Hope that makes sense to you, otherwise I'll just drop the weak pointer stuff from the new testcase.